fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Utils.java

```diff
@@ -536,4 +569,273 @@ private static <T extends ScalarOperator> void collect(ScalarOperator root, Clas
 
         root.getChildren().forEach(child -> collect(child, clazz, output));
     }
+
+    public static void getRelatedMvs(List<Table> tablesToCheck, Set<MaterializedView> mvs) {
+        Set<MvId> newMvIds = Sets.newHashSet();
+        for (Table table : tablesToCheck) {
+            Set<MvId> mvIds = table.getRelatedMaterializedViews();
+            if (mvIds != null && !mvIds.isEmpty()) {
+                newMvIds.addAll(mvIds);
+            }
+        }
+        if (newMvIds.isEmpty()) {
+            return;
+        }
+        List<Table> newMvs = Lists.newArrayList();
+        for (MvId mvId : newMvIds) {
+            Database db = GlobalStateMgr.getCurrentState().getDb(mvId.getDbId());
+            if (db == null) {
+                continue;
+            }
+            Table table = db.getTable(mvId.getId());
+            if (table == null) {
+                continue;
+            }
+            newMvs.add(table);
+            mvs.add((MaterializedView) table);
+        }
+        getRelatedMvs(newMvs, mvs);
+    }
+
+    // get all ref tables within and below root
+    public static List<Table> getAllTables(OptExpression root) {
```

Comment:
Duplicated with `AnalyzerUtils::collectAllTableRelation`?

Reply:
no, they are not the same. collectAllTableRelation is for TableRelation. getAllTables is for OptExpression.
They are used in different phase.